build_image.sh

```diff
@@ -87,11 +87,24 @@ generate_onie_installer_image()
           $ONIE_INSTALLER_PAYLOAD
 }
 
+generate_device_list()
+{
+    # Generate asic-specific ONIE device list
+    rm -rf ./installer/x86_64/devices/
+    mkdir -p ./installer/x86_64/devices/
+    for d in `find -L ./device  -maxdepth 2 -mindepth 2 -type d`; do
+        if [ -f $d/platform_asic ] && grep -Fxq "$CONFIGURED_PLATFORM" $d/platform_asic; then
+            echo "${d##*/}" >> ./installer/x86_64/devices/platforms_asic;
```

Comment:
>./installer/x86_64/devices/platforms_asic [](http://example.com/codeflow?start=31&length=41)

This filename could be replace by a function parameter, and onie/aboot will call the function with different argument.

Then you don't need the extra step `cp ./installer/x86_64/devices/platforms_asic .platforms_asic`